static/js/src/advantage/api/contracts.test.ts

```diff
@@ -1,25 +1,42 @@
+import {
+  contractTokenFactory,
+  lastPurchaseIdsFactory,
+  userSubscriptionFactory,
+} from "advantage/tests/factories/api";
 import fetch from "jest-fetch-mock";
 
-import { getContractToken, getUserSubscriptions } from "./contracts";
+import {
+  getContractToken,
+  getLastPurchaseIds,
+  getUserSubscriptions,
+} from "./contracts";
 
 describe("contracts", () => {
   afterEach(() => {
     fetch.resetMocks();
   });
 
-  it("can get user subscriptions", () => {
-    const contractData = { data: { test: "contract" } };
+  it("can get user subscriptions", async () => {
+    const contractData = { data: userSubscriptionFactory.buildList(2) };
```

Comment:
Just a personal opinion, I don't see a clear benefit of using yet another layer of abstraction here. This line before the change was better at explaining how this code works.

Reply:
We use factories based on the typescript API types to produce API data as we expect it to. I'm very happy to talk with you about the benefits of using a factory lib, in fact I'm going to do a masterclass which will touch on this topic a bit. However, I'm not sure this PR is the right place to explain that in detail.